.github/workflows/build-and-test.yml

```diff
@@ -178,6 +178,10 @@ jobs:
           git diff --exit-code ':!*go.sum' || (echo 'Generated code is out of date, please run "make generate" and commit the changes in this PR.' && exit 1)
       - name: Impi
         run: make -j2 goimpi
+      - name: GCI
+        run: |
+          make -j2 gogci
```

Comment:
Could we run this instead as part of golangci-lint? So, instead of adding a new Makefile target we just add it to the list here https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/09e2f9bca83cd308b81398141374e296b7901d83/.golangci.yml#L119-L120